### PR TITLE
[ci] Move CI to GitHub Actions (fixes #215)

### LINKED ISSUE #215: [ci] Move CI from Travis to GitHub Actions
Travis is putting much much stricter limits on free-tier use of its CI services.

https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

> For those of you who have been building on public repositories (on travis-ci.com, with no paid subscription), we will upgrade you to our trial (free) plan with a 10K credit allotment (which allows around 1000 minutes in a Linux environment).

> We will be offering an allotment of OSS minutes that will be reviewed and allocated on a case by case basis. Should you want to apply for these credits please open a request with Travis CI support stating that you’d like to be considered for the OSS allotment.

The blog post also makes it sound like macOS jobs will no longer be available for free-tier users, but honestly the blog post's discussion of that is very confusing.

This project's CI should just be moved to GitHub Actions, to not have to deal with this.

### PR DESCRIPTION
**Pull request by jameslamb**
This PR attempts to fix #215. See that issue and the links in it for details. Basically, Travis is becoming a lot harder for open source projects to use, and I think GitHub Actions is mature enough that we can move this project to it.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (austin3dickey) [inline .github/workflows/ci.yml:16]: cute

**Review by austin3dickey — APPROVED** [review]
testing whether the transient-failed build will block merge even if I approve

**Comment by austin3dickey** [comment]
Yep LGTM! Looks like you just need to rebuild because it timed out trying to pull data.table from CRAN in one of the tests. Great work!

**Comment by jameslamb** [comment]
> Looks like you just need to rebuild because it timed out trying to pull data.table from CRAN in one of the tests

:grimacing: hopefully that's just a fluke. If it becomes a real problem we could probably change the timeout or start caching those packages or something

**Comment by jameslamb** [comment]
I just changed the branch rules to require that all GitHub Actions jobs pass before a PR can be merged

image

File: .github/workflows/ci.yml
Comment (jameslamb) [reply .github/workflows/ci.yml:16]: thank you, thank you

**Pull request merged**
